galaxyxml/tool/parameters/__init__.py

```diff
@@ -363,29 +386,19 @@ def __init__(self, name, **kwargs):
             kwargs = dict([("name", name)] + list(kwargs.items()))
 
         # Handle positional parameters
-        if "positional" in kwargs and kwargs["positional"]:
-            self.positional = True
-        else:
-            self.positional = False
-
-        if "num_dashes" in kwargs:
-            self.num_dashes = kwargs["num_dashes"]
-            del kwargs["num_dashes"]
-        else:
-            self.num_dashes = 0
-
-        self.space_between_arg = " "
+        self.positional = kwargs.pop("positional", False)
+        self.num_dashes = kwargs.pop("num_dashes", 0)
+        self.space_between_arg = kwargs.pop("space_between_arg", " ")
```

Comment:
I only see `num_dashes` removed before, so popping the others is a change. is that right?

Reply:
Unsure. Guess the safe thing to do would be: 

```suggestion
        self.positional = kwargs.get("positional", False)
        self.num_dashes = kwargs.pop("num_dashes", 0)
        self.space_between_arg = kwargs.get("space_between_arg", " ")
```

Reply:
yeah I like that better, agreed.